extensions/stats/plugin.cc

```diff
@@ -70,24 +70,24 @@ void map_node(IstioDimensions& instance, bool is_source,
       auto version = version_iter ? version_iter->value() : nullptr;
       FB_ASSIGN(source_version, version);
 
-      auto canonical_name =
-          source_labels->LookupByKey("service.istio.io/canonical-name");
+      auto canonical_name = source_labels->LookupByKey(
+          ::Wasm::Common::kCanonicalServiceLabelName.data());
       auto name =
           canonical_name ? canonical_name->value() : node.workload_name();
       FB_ASSIGN(source_canonical_service, name);
 
-      auto rev =
-          source_labels->LookupByKey("service.istio.io/canonical-revision");
+      auto rev = source_labels->LookupByKey(
+          ::Wasm::Common::kCanonicalServiceRevisionLabelName.data());
       if (rev) {
         FB_ASSIGN(source_canonical_revision, rev->value());
       } else {
-        instance[source_canonical_revision] = "latest";
+        instance[source_canonical_revision] = ::Wasm::Common::kLatest.data();
```

Comment:
This is expecting a string. Can you just pass a string view to avoid the scan? The compiler should auto convert from string_view to string.

Reply:
```
extensions/stats/plugin.cc:84:45: error: no viable overloaded '='
        instance[source_canonical_revision] = ::Wasm::Common::kLatest;
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/basic_string.h:665:7: note: candidate function not viable: no known conversion from 'const absl::string_view' to 'const std::__cxx11::basic_string<char>' for 1st argument
      operator=(const basic_string& __str)
      ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/basic_string.h:704:7: note: candidate function not viable: no known conversion from 'const absl::string_view' to 'const char *' for 1st argument
      operator=(const _CharT* __s)
      ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/basic_string.h:715:7: note: candidate function not viable: no known conversion from 'const absl::string_view' to 'char' for 1st argument
      operator=(_CharT __c)
      ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/basic_string.h:732:7: note: candidate function not viable: no known conversion from 'const absl::string_view' to 'std::__cxx11::basic_string<char>' for 1st argument
      operator=(basic_string&& __str)
      ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/basic_string.h:795:7: note: candidate function not viable: no known conversion from 'const absl::string_view' to 'initializer_list<char>' for 1st argument
      operator=(initializer_list<_CharT> __l)
```

Reply:
I'm beginning to have concerns about `absl::string_view` as it requires `.data()` calls when using with FlatBuffers (as far as I can tell).

From the docs:

> Note that `string_view::data()` may contain embedded nul
  // characters, but the returned buffer may or may not be NUL-terminated;
  // therefore, do not pass `data()` to a routine that expects a NUL-terminated
  // string.

I think we should use `char[]` here.

Reply:
Not a big deal. It'll get much cleaner with C++17.

Reply:
sorry, what do you think we need here to get this in?